agent/src/main/java/org/openremote/agent/protocol/mqtt/MQTTProtocol.java

```diff
@@ -107,7 +118,15 @@ protected MQTT_IOClient doCreateIoClient() throws Exception {
             lastWill = new MQTTLastWill(topic, payload, retain);
         }
 
-        return new MQTT_IOClient(agent.getClientId().orElseGet(UniqueIdentifierGenerator::generateId), host, port, agent.isSecureMode().orElse(false), !agent.isResumeSession().orElse(false), agent.getUsernamePassword().orElse(null), websocketURI, lastWill);
+		//It's fine if they're null, they're not going to be used when creating the client
+	    TrustManagerFactory trustManagerFactory = null;
+		KeyManagerFactory keyManagerFactory = null;
+		if(agent.isSecureMode().orElse(false)){
+			trustManagerFactory = keyStoreService.getTrustManagerFactory();
+			keyManagerFactory = keyStoreService.getKeyManagerFactory(agent.getRealm()+"."+agent.getCertificateAlias().orElseThrow());
```

Comment:
It could be possible to have a custom trust cert but not want client auth and maybe vice versa so throwing exceptions if not set implies it is required.